File: src/pocketcloud/cloud/event/impl/server/ServerSendCommandEvent.php
<?php

namespace pocketcloud\cloud\event\impl\server;

use pocketcloud\cloud\event\Cancelable;
use pocketcloud\cloud\event\CancelableTrait;
use pocketcloud\cloud\server\CloudServer;

class ServerSendCommandEvent extends ServerEvent implements Cancelable {
    use CancelableTrait;

    public function __construct(
        CloudServer $server,
        private readonly string $commandLine
    ) {
        parent::__construct($server);
    }

    public function getCommandLine(): string {
        return $this->commandLine;
    }
}

File: cloud-core/src/main/java/de/pocketcloud/cloud/event/impl/server/ServerSendCommandEvent.java
package de.pocketcloud.cloud.event.impl.server;

import de.pocketcloud.cloud.event.Cancelable;
import de.pocketcloud.cloud.server.CloudServer;
import lombok.Getter;

@Getter
public class ServerSendCommandEvent extends ServerEvent implements Cancelable {

    private final String commandLine;

    public ServerSendCommandEvent(CloudServer server, String commandLine) {
        super(server);
        this.commandLine = commandLine;
    }
}
